url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/data/gathering/dataSource/reuters/URLExploreJob.java

```diff
@@ -0,0 +1,67 @@
+package darwInvest.data.gathering.dataSource.reuters;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Scanner;
+import darwin.work.Job;
+
+/**
+ * This object abstracts away the parallelism of URL exploration
+ * to speed up the data gathering process significantly
+ * 
+ * @author Kevin Dolan
+ */
+public class URLExploreJob implements Job  {
+	
+	@Override
+	public Object run(Object parameters, HashMap<String, Object> shared) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.